File: Java/55.跳跃游戏.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int reach = 0;
        for(int i = 0; i < n && i <= reach; i++){
            reach = Math.max(reach, i + nums[i]);
            if(reach >= n - 1) return true;
        }
        return false;
    }
}

File: Python/55.跳跃游戏.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n = len(nums)
        reach = 0
        i = 0
        while i < n and i <=reach:
            reach = max(reach, i + nums[i])
            i = i + 1
            if reach >= n-1:
                return True
        return False